drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -0,0 +1,480 @@
+# -*- coding: utf-8 -*-
+#
+
+# Imports ###########################################################
+
+import json
+import webob
+import copy
+import urllib
+
+from xblock.core import XBlock
+from xblock.fields import Scope, String, Dict, Float, Boolean
+from xblock.fragment import Fragment
+from xblockutils.resources import ResourceLoader
+from xblockutils.settings import XBlockWithSettingsMixin, ThemableXBlockMixin
+
+from .utils import _  # pylint: disable=unused-import
+from .default_data import DEFAULT_DATA
+
+
+# Globals ###########################################################
+
+loader = ResourceLoader(__name__)
+
+
+# Classes ###########################################################
+
+@XBlock.wants('settings')
+class DragAndDropBlock(XBlock, XBlockWithSettingsMixin, ThemableXBlockMixin):
+    """
+    XBlock providing a Drag and Drop question
+    """
+    display_name = String(
+        display_name=_("Title"),
+        help=_("The title of the Drag and Drop that is displayed to the user"),
+        scope=Scope.settings,
+        default=_("Drag and Drop"),
+    )
+
+    show_title = Boolean(
+        display_name=_("Show title"),
+        help=_("Display the title to the user?"),
+        scope=Scope.settings,
+        default=True,
+    )
+
+    question_text = String(
+        display_name=_("Question text"),
+        help=_("The question text (and/or instructions) shown to the user"),
+        scope=Scope.settings,
+        default="",
+    )
+
+    show_question_header = Boolean(
+        display_name=_("Show \"Question\" heading"),
+        help=_("Display the \"Question\" heading above the question/instructions?"),
+        scope=Scope.settings,
+        default=True,
+    )
+
+    weight = Float(
+        display_name=_("Weight"),
+        help=_("This is the maximum score that the user receives when he/she successfully completes the problem"),
+        scope=Scope.settings,
+        default=1,
+    )
+
+    item_background_color = String(
+        display_name=_("Item background color"),
+        help=_(
+            "Background color to use for draggable items. "
+            "Defaults to color specified by current theme, or blue if no theme is set."
+        ),
+        scope=Scope.settings,
+        default="",
+    )
+
+    item_text_color = String(
+        display_name=_("Item text color"),
+        help=_(
+            "Text color to use for draggable items. "
+            "Defaults to color specified by current theme, or white if no theme is set."
+        ),
+        scope=Scope.settings,
+        default="",
+    )
+
+    data = Dict(
+        display_name=_("Drag and Drop"),
+        help=_("JSON spec as generated by the builder"),
+        scope=Scope.content,
+        default=DEFAULT_DATA,
+    )
+
+    item_state = Dict(
+        help=_("How the student has interacted with the problem"),
```

Comment:
@pdesjardins This field is for internal use only; it's not available in any UI, and it's also not part of any (public) API. So just like the help string for `data`, the help string for this field is only used to convey information about the field to developers.

> This looks like a description of the state of a problem after a learner has attempted it. Is that right?

Yes, if "has attempted it" refers to any type of interaction (when a learner drops an item on the appropriate zone, information about the position of the item is added to `item_state`; it doesn't matter if the learner completed the problem by performing this action).

I updated the string to be more specific about the type of information that `item_state` stores; it now reads: "Information about current positions of items that a learner has dropped on the target image." Taking into account the considerations above, please let me know if you have any suggestions for further improvement.